### DISCUSSION TITLE
Is there a way to add a progress bar to my components?

### DISCUSSION THREAD

**Discussion by capjjs** [livewire/livewire#986] [4. General]
I have a wire:click that calls a method recalculateStats. This method is a bit expensive and takes some time to complete and I want to show a progress bar on the front end side. Is there a way to accomplish that with livewire?

**Comment by mblarsen** [comment]
What would your measure of progress be? I mean how would the progress bar know when and with what amount to update/

Often the user is happen simply to know that something is happening. For that you can use [Loading states](https://laravel-livewire.com/docs/loading-states#toggling-elements), however, if it takes a long time you are right some kind of progress bar would be nice.

**Reply by capjjs** [reply]
On the back-end I'm looping thru a chunk of data and the progress would basically be  
(processed rows) / (total rows)

I'm looking for something like the **XMLHttpRequest: progress event**

Here's a sample of what I'm trying to accomplish. 
_[https://zinoui.com/blog/ajax-request-progress-bar](url)_

**Comment by taropogi** [comment]
I also have the same question.

**Comment by tawfek** [comment]
same question , did you find any solve for this problem ?
